README.md
kibana-ng Cookbook
================

[![Build Status](https://travis-ci.org/vkhatri/chef-kibana-ng.svg?branch=master)](https://travis-ci.org/vkhatri/chef-kibana-ng)

This is a [Chef] cookbook to manage [Kibana].

This cookbook was heavily inspired from simple-kibana cookbook maintained by jsirex.


## Repository

https://github.com/vkhatri/chef-kibana-ng


## Supported Version

This cookbook was tested for Kibana v4.0.2+ x86_64 version.


## Supported OS

This cookbook was tested on Amazon Linux & Ubuntu and expected to work on other RHEL platforms.


## Recipes

- `kibana-ng::default` - default recipe (used for run_list)

- `kibana-ng::install` - wrapper recipe to install kibana

- `kibana-ng::config` - configures kibana

- `kibana-ng::nginx` - configure kibana nginx proxy

- `kibana-ng::service` - configure kibana service

- `kibana-ng::tarball` - install kibana from tarball

- `kibana-ng::user` - install user / group for kibana


## Core Attributes


* `default['kibana']['version']` (default: `4.1.2`): kibana version

* `default['kibana']['user']` (default: `kibana`): kibana user

* `default['kibana']['group']` (default: `kibana`): kibana group

* `default['kibana']['setup_user']` (default: `true`): setup kibana user group

* `default['kibana']['service_name']` (default: `kibana`): kibana service name

* `default['kibana']['service_type']` (default: `sysv`): kibana service type, currently supports only `sysv`

* `default['kibana']['install_method']` (default: `source`): kibana install method, currently supports only `source`

* `default['kibana']['umask']` (default: `0022`): execute resource umask

* `default['kibana']['dir_mode']` (default: ``): kibana file/dir resource mode

* `default['kibana']['parent_dir']` (default: `/opt/kibana`): kibana parent/home directory for different versions

* `default['kibana']['log_dir']` (default: `/var/log/kibana`): kibana service log

* `default['kibana']['conf_dir']` (default: `/etc/kibana`): kibana yaml configuration file directory

* `default['kibana']['install_dir']` (default: `/opt/kibana/kibana`): kibana install / current version location, symlink to current version location

* `default['kibana']['daemon']` (default: `/opt/kibana/kibana/bin/kibana`): kibana bin location

* `default['kibana']['conf_file']` (default: `/etc/kibana/kibana.yml`): kibana configuration file

* `default['kibana']['log_file']` (default: `/var/log/kibana/kibana.log`): kibana log file location

* `default['kibana']['pid_file']` (default: `/opt/kibana/kibana/kibana.pid`): kibana service pid file location

* `default['kibana']['auth_users']` (default: `{guest:hash}`): nginx auth users hash

* `default['kibana']['nginx_port']` (default: `{guest:hash}`): nginx auth users hash

* `default['kibana']['nginx_auth_file']` (default: `calculated`): nginx auth users file

* `default['kibana']['nginx_port']` (default: `80`): nginx proxy port


## Configuration File kibana.yml Attributes

* `default['kibana']['config']['port']` (default: `5601`): kibana port

* `default['kibana']['config']['host']` (default: `127.0.0.1`): kibana host address

* `default['kibana']['config']['pid_file']` (default: `/opt/kibana/kibana/kibana.pid`): kibana pid file location

* `default['kibana']['config']['elasticsearch_url']` (default: `http://localhost:9200`): elasticsearch url

* `default['kibana']['config']['kibana_index']` (default: `.kibana`): elasticsearch index

* `default['kibana']['config']['default_app_id']` (default: `discover`):

* `default['kibana']['config']['request_timeout']` (default: `300_000`):

* `default['kibana']['config']['shard_timeout']` (default: `0`):

* `default['kibana']['config']['verify_ssl']` (default: `true`):

* `default['kibana']['config']['bundled_plugin_ids']` (default: `[]`):

For more attribute info, visit below links:

https://github.com/elastic/kibana/blob/master/src/server/config/kibana.yml

https://www.elastic.co/guide/en/kibana/current/_setting_kibana_server_properties.html


## Contributing

1. Fork the repository on Github
2. Create a named feature branch (like `add_component_x`)
3. Write your change
4. Write tests for your change (if applicable)
5. Run the tests (`rake & rake knife`), ensuring they all pass
6. Write new resource/attribute description to `README.md`
7. Write description about changes to PR
8. Submit a Pull Request using Github


## Copyright & License

Authors:: Virender Khatri and [Contributors]

<pre>
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
</pre>


[Chef]: https://www.chef.io/
[Kibana]: https://github.com/elastic/kibana/
[Contributors]: https://github.com/vkhatri/chef-kibana-ng/graphs/contributors


CHANGELOG.md
kibana-ng CHANGELOG
===================

This file is used to list changes made in each version of the kibana-ng cookbook.

0.2.3
-----

- Virender Khatri - bump kibana version to 4.1.2

- Virender Khatri - fix rubocop check

- Virender Khatri - rename source install method to tarball

- Virender Khatri - add tarball sha256sums to helper method

0.1.9
-----

- Virender Khatri - issue #1, bump kibana version to 4.1.1

0.1.8
-----

- Virender Khatri - bind kibana host to localhost

- Virender Khatri - add kibana service notify on config change

- Virender Khatri - added kibana nginx proxy

- Virender Khatri - bump kibana version to v4.1.0

- Virender Khatri - store multiple version source checksum

- Virender Khatri - added debian platform family support, upstart config

- Virender Khatri - updated README and fixed typos

0.1.1
-----

- Virender Khatri - Updated README and minor changes

0.1.0
-----

- Virender Khatri - Initial release of kibana-ng

- - -
Check the [Markdown Syntax Guide](http://daringfireball.net/projects/markdown/syntax) for help with Markdown.

The [Github Flavored Markdown page](http://github.github.com/github-flavored-markdown/) describes the differences between markdown on github and standard markdown.


attributes/default.rb
default['kibana']['version'] = '4.1.2'

default['kibana']['user'] = 'kibana'
default['kibana']['group'] = 'kibana'
default['kibana']['setup_user'] = true

default['kibana']['service_name'] = 'kibana'
default['kibana']['service_type'] = 'sysv' # options: sysv
default['kibana']['notify_restart'] = true

default['kibana']['install_method'] = 'tarball' # options: tarball

default['kibana']['auth_users']['guest'] = '$apr1$cA/eVUgT$aIoWUPwV5uONJoYslb7lg0'

default['kibana']['umask']        = '0022'
default['kibana']['dir_mode']     = '0755'
default['kibana']['parent_dir']   = '/opt/kibana'
default['kibana']['log_dir']      = '/var/log/kibana'
default['kibana']['conf_dir']     = '/etc/kibana'
default['kibana']['install_dir']  = ::File.join(node['kibana']['parent_dir'], 'kibana')
default['kibana']['daemon']       = ::File.join(node['kibana']['install_dir'], 'bin', 'kibana')

default['kibana']['conf_file'] = ::File.join(node['kibana']['conf_dir'], 'kibana.yml')
default['kibana']['log_file'] = ::File.join(node['kibana']['log_dir'], 'kibana.log')
default['kibana']['pid_file'] = ::File.join(node['kibana']['parent_dir'], 'kibana.pid')

default['kibana']['nginx_auth_file'] = ::File.join(node['nginx']['dir'], 'htpasswd.kibana')
default['kibana']['nginx_port'] = 80

default['nginx']['default_site_enabled'] = false
